packages/pluggableWidgets/line-chart-native/src/components/LineChart.tsx

```diff
@@ -0,0 +1,168 @@
+                );
+
+            return (
+                <VictoryGroup key={index}>
+                    {markers && (!seriesStyle?.markers?.display || seriesStyle.markers.display !== "onTop")
+                        ? markers
+                        : null}
+                    <VictoryLine style={seriesStyle?.line} data={dataPoints} interpolation={interpolation} />
+                    {markers && seriesStyle?.markers?.display === "onTop" ? markers : null}
+                </VictoryGroup>
+            );
+        });
+    }, [dataTypesResult, series, style]);
+
+    const [chartDimensions, setChartDimensions] = useState<{ height: number; width: number }>();
+
+    const updateChartDimensions = useCallback(
+        (event: LayoutChangeEvent) => {
+            const { height, width } = event.nativeEvent.layout;
+            setChartDimensions({ height, width });
+        },
+        [setChartDimensions]
+    );
+
+    return (
+        <View style={style.container}>
+            {dataTypesResult instanceof Error ? (
+                <Text>{dataTypesResult.message}</Text>
+            ) : (
+                <Fragment>
+                    <View onLayout={updateChartDimensions} style={{ flex: 1 }}>
+                        {chartDimensions ? (
+                            <VictoryChart
+                                height={chartDimensions?.height}
+                                width={chartDimensions?.width}
+                                padding={style.chart?.padding}
+                                scale={
+                                    dataTypesResult
+                                        ? {
+                                              x: dataTypesResult.x === "number" ? "linear" : "time",
+                                              y: dataTypesResult.y === "number" ? "linear" : "time"
+                                          }
+                                        : undefined
+                                }
+                                style={style.chart}
+                            >
+                                <VictoryAxis
+                                    style={style.xAxis}
+                                    axisLabelComponent={
+                                        xAxisLabel ? (
+                                            <VictoryLabel
+                                                dx={style.xAxis?.axisLabel?.horizontalOffset}
+                                                dy={style.xAxis?.axisLabel?.verticalOffset}
+                                            />
+                                        ) : (
+                                            undefined
+                                        )
+                                    }
+                                    label={xAxisLabel}
+                                    orientation={"bottom"}
+                                    {...(series[0]?.xFormatter ? { tickFormat: series[0].xFormatter } : undefined)}
```

Comment:
can we extract `series[0]` somewhere above? Then we can re-use the variable in this line and 136

Reply:
Extracted outside return statement